boring/src/x509/tests.rs

```diff
@@ -240,6 +240,7 @@ fn x509_builder() {
 
     assert!(pkey.public_eq(&x509.public_key().unwrap()));
     assert!(x509.verify(&pkey).unwrap());
+    assert_eq!(x509.extensions().unwrap().len(), 6);
```

Comment:
> `6`

We should probably collect all of the extensions to a collection and call `append_extension` using an iterator, so that we don’t have this magic number.